contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -432,52 +434,55 @@ tsql_get_functiondef(PG_FUNCTION_ARGS)
 		appendStringInfoString(&buf, "(");
 	
 	/* we will not pfree name because as we can see name = NameStr(proc->proname) 
-         * here we are not allocating extra space for name, we’re just using proc-> proname.
-         * also at the end, we’re releasing proctup (that will free proc->proname).  
-         */
-	pfree(nsp);
-	if (nnsp)
-		pfree(nnsp);
-        
+	 * here we are not allocating extra space for name, we’re just using proc-> proname.
+	 * also at the end, we’re releasing proctup (that will free proc->proname).  
+	 */
+
 	tmp = SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_probin, &isnull);
-        number_args = proc->pronargs;
-        if(isfunction) number_args++;
+
+	if (!isnull)
+		probin_c = TextDatumGetCString(tmp);
+	if(!probin_c || probin_c[0] != '{')
+		PG_RETURN_NULL();
+	
+	number_args = proc->pronargs;
+	if (isfunction)
+		number_args++;
 	/* Return NULL for the definition if procedure language is not pltsql. */
 	if(strcmp(get_language_name(proc->prolang, false), "pltsql") != 0)
 		PG_RETURN_NULL();
-       	probin_json_reader(tmp, &typmod_arr, number_args);
+
+	probin_json_reader(cstring_to_text(probin_c), &typmod_arr, number_args);
+	pfree(probin_c);
 	(void) tsql_print_function_arguments(&buf, proctup, false, true, &typmod_arr, &has_tvp);
 	/* TODO: In case of Table Valued Functions, return NULL. */
 	if (has_tvp)
 		PG_RETURN_NULL();
+
 	if(isfunction || proc->pronargs > 0)
 		appendStringInfoString(&buf, ")");
 	if (isfunction)
 	{
 		appendStringInfoString(&buf, " RETURNS ");
 		tsql_print_function_rettype(&buf, proctup, &typmod_arr, number_args);
 	}
-        if(typmod_arr)
+	if(typmod_arr)
 		pfree(typmod_arr);
  
 	/* Emit some miscellaneous options on one line */
-	oldlen = buf.len;
-
-        if (proc->proisstrict)
+	if (proc->proisstrict)
 		appendStringInfoString(&buf, " WITH RETURNS NULL ON NULL INPUT");
 
 	/* And finally the function definition ... */
 	(void) SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_prosqlbody, &isnull);
 	
-        appendStringInfoString(&buf, " AS ");
+	appendStringInfoString(&buf, " AS ");
 	tmp = SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_prosrc, &isnull);
 	prosrc = TextDatumGetCString(tmp); 
 	appendStringInfoString(&buf, prosrc);
 
 	ReleaseSysCache(proctup);
 
-        pfree(prosrc);
```

Comment:
why is this removed?

Reply:
`prosrc` is const pointer, We should not free const pointers.